src/test_l2_hdmi_cec_driver.c

```diff
@@ -1,77 +1,394 @@
 */
-void test_l2_hdmi_cec_driver (void)
+
+void test_l2_hdmi_cec_driver_GetDefaultLogicalAddress(void)
 {
-	UT_FAIL(This function needs to be implemented!); 
+    UT_LOG("Entering test_l2_hdmi_cec_driver_GetDefaultLogicalAddress...");
+
+    int handle;
+    int logicalAddress;
+    HDMI_CEC_STATUS status;
+
+    // Step 1: Call the pre-requisite API HdmiCecOpen() to initialize the HDMI CEC HAL.
+    status = HdmiCecOpen(&handle);
+    UT_LOG("Invoking HdmiCecOpen() with valid handle. Return status: %d", status);
+    UT_ASSERT_EQUAL(status, HDMI_CEC_IO_SUCCESS);
+
+    // Step 2: Call the API HdmiCecGetLogicalAddress() with the handle obtained from HdmiCecOpen() and a pointer to an integer to store the logical address.
+    status = HdmiCecGetLogicalAddress(handle, &logicalAddress);
+    UT_LOG("Invoking HdmiCecGetLogicalAddress() with handle from HdmiCecOpen() and valid logicalAddress pointer. Return status: %d, Logical Address: %d", status, logicalAddress);
+    UT_ASSERT_EQUAL(status, HDMI_CEC_IO_SUCCESS);
+
+    // Step 3: Check that the logical address returned is 0x0F, which is the default logical address for sink devices when no logical address has been added or removed.
+    UT_ASSERT_EQUAL(logicalAddress, 0x0F);
+
+    // Step 4: Call the post-requisite API HdmiCecClose() with the handle obtained from HdmiCecOpen() to uninitialize the HDMI CEC HAL.
+    status = HdmiCecClose(handle);
+    UT_LOG("Invoking HdmiCecClose() with handle from HdmiCecOpen(). Return status: %d", status);
+    UT_ASSERT_EQUAL(status, HDMI_CEC_IO_SUCCESS);
+
+    UT_LOG("Exiting test_l2_hdmi_cec_driver_GetDefaultLogicalAddress...");
+}
+
+/**
+* @brief Test the sequence of operations: HdmiCecOpen, HdmiCecAddLogicalAddress, HdmiCecGetLogicalAddress, and HdmiCecClose
+*
+* This function tests the sequence of operations: HdmiCecOpen, HdmiCecAddLogicalAddress, HdmiCecGetLogicalAddress, and HdmiCecClose. It uses the CUnit framework's assertion macros to check that each operation returns the expected status code. It also checks that the logical address returned by HdmiCecGetLogicalAddress matches the address that was set by HdmiCecAddLogicalAddress.
+*
+* **Test Group ID:** 02@n
+* **Test Case ID:** 002@n
+*
+* **Test Procedure:**
+* Refer to UT specification documentation [l2_hdmi_cec_driver_test_specification.md](l2_hdmi_cec_driver_test_specification.md)
+*/
+
+void test_l2_hdmi_cec_driver_AddAndGetLogicalAddress(void)
+{
+    UT_LOG("Entering test_l2_hdmi_cec_driver_AddAndGetLogicalAddress...");
+
+    int handle = 0;
+    int logicalAddress = 0;
+    HDMI_CEC_STATUS status;
+
+    // Step 1: Call HdmiCecOpen
+    status = HdmiCecOpen(&handle);
+    UT_LOG("Invoking HdmiCecOpen(). Status: %d, Handle: %d", status, handle);
+    UT_ASSERT_EQUAL(status, HDMI_CEC_IO_SUCCESS);
+    UT_ASSERT_NOT_EQUAL(handle, 0);
+
+    // Generate a random logical address between 0x00 and 0x0F
+    srand(time(NULL));
```

Comment:
need to update the test requirement, since the current one says "Set up a legitimate logical address (0x00 to 0x0F) for the DUT". Will update the test requirement and regenarate